### PR TITLE
Update google.rst to clarify location v. region

### PR DESCRIPTION
**Pull request by moschetti**
Add details and links which clarify the difference between location (Cloud Life Sciences API location) and region (Compute Engine region).

### CODE REVIEW

File: docs/google.rst [hunk 1]
```diff
@@ -456,6 +460,7 @@ Example::
 
     google {
         project = 'your-project-id'
+        location = 'europe-west2'
         zone = 'europe-west1-b'
```
Comment (pditommaso) [inline docs/google.rst]: Then, the `location` could be omitted from the example snippet. Do you agree?

File: docs/google.rst [hunk 2]
```diff
@@ -572,6 +577,7 @@ For example::
...
```
Comment (pditommaso) [inline docs/google.rst]: Same as before.

File: docs/google.rst (hunk 1 above)
Comment (moschetti) [reply docs/google.rst]: agreed. I'll remove the example snippets but leave the definition part and submit an updated pr

Comment (pditommaso) [reply docs/google.rst]: thanks!

**Pull request merged**